src/plugins/event-plugins/ResourcePlugin.ts

```diff
@@ -49,73 +52,67 @@ export class ResourcePlugin extends InternalPlugin {
     }
 
     performanceEntryHandler = (list: PerformanceObserverEntryList): void => {
-        this.recordPerformanceEntries(list.getEntries());
-    };
-
-    recordPerformanceEntries = (list: PerformanceEntryList) => {
-        const recordAll: PerformanceEntry[] = [];
-        const sample: PerformanceEntry[] = [];
+        for (const e of list.getEntries()) {
+            const entry = e as PerformanceResourceTimingPolyfill;
+            const { name, initiatorType } = entry;
 
-        list.filter((e) => e.entryType === RESOURCE)
-            .filter((e) => !this.config.ignore(e))
-            .forEach((event) => {
-                const { name, initiatorType } =
-                    event as PerformanceResourceTiming;
-                const type = getResourceFileType(name, initiatorType);
-                if (this.config.recordAllTypes.includes(type)) {
-                    recordAll.push(event);
-                } else if (this.config.sampleTypes.includes(type)) {
-                    sample.push(event);
-                }
-            });
-
-        // Record all events for resources in recordAllTypes
-        recordAll.forEach((r) =>
-            this.recordResourceEvent(r as PerformanceResourceTiming)
-        );
+            // (1) Ignore by custom callback.
+            // (2) Ignore calls to PutRumEvents (i.e., the CloudWatch RUM data plane),
+            // otherwise we end up in an infinite loop of recording PutRumEvents.
+            if (
+                this.config.ignore(e) ||
+                isPutRumEventsCall(name, this.context.config.endpointUrl.host)
```

Comment:
issue: Aren't fetch and xhr requests already ignored by the default config?

Reply:
Yes you are right. This is old logic, but I'm not sure that this is needed either until we add support for Fetch/Xhr.

Reply:
I didn't realize this was the original logic. I agree that, with respect to this PR, we should leave as-is.

Regarding the documentation change, "Ignore by custom callback" isn't clear. Please clean up the comment and move it back inside the if statement.

Reply:
Restored the original comment.